### PR TITLE
xmrig: Various tweaks on the formula

### PR DESCRIPTION
**Pull request by rtau**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/xmrig.rb [hunk 1]
```diff
@@ -11,16 +11,23 @@ class Xmrig < Formula
     sha256 "3a274f5c98cd7fb6946e208bf2b4e1a5309ce43f92e937208e188324bef3a937" => :el_capitan
   end
 
+  option "without-httpd", "Build without built in http server and API"
+  option "without-aeon", "Disable CryptoNight-Lite support"
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: I'm not sure I see the value of the `without-aeon` option: it does not reduce the list of dependencies, so it's kind of a freebie to keep it in

File: Formula/xmrig.rb [hunk 2]
```diff
@@ -11,16 +11,23 @@ class Xmrig < Formula
...
+
   depends_on "cmake" => :build
-  depends_on "libmicrohttpd"
+  depends_on "libmicrohttpd" if build.with? "httpd"
```
Comment (DomT4) [inline Formula/xmrig.rb]: `if build.with?` deps are grim outside of `def install`. If you absolutely need it the option should be called `"without-libmicrohttpd"`.

File: Formula/xmrig.rb [hunk 3]
```diff
@@ -11,16 +11,23 @@ class Xmrig < Formula
...
+    args = []
+    args << "-DWITH_HTTPD=OFF" if build.without? "httpd"
+    args << "-DWITH_AEON=OFF" if build.without? "aeon"
     mkdir "build" do
-      system "cmake", "..", "-DUV_LIBRARY=#{Formula["libuv"].opt_lib}/libuv.a", *std_cmake_args
+      system "cmake", "..", *args, *std_cmake_args
       system "make"
       bin.install "xmrig"
     end
+    share.install "src/config.json"
```
Comment (DomT4) [inline Formula/xmrig.rb]: `pkgshare`.

File: Formula/xmrig.rb (hunk 1 above)
Comment (rtau) [reply Formula/xmrig.rb]: That just replicate the additional cmake options provided in upstream wiki. Personally I don't use it neither.

https://github.com/xmrig/xmrig/wiki/OS-X-Build

File: Formula/xmrig.rb (hunk 2 above)
Comment (rtau) [reply Formula/xmrig.rb]: I have not much concern on naming that option to be longer or shorter, but that name is sort of same as the cmake option.

**Comment by fxcoudert** [comment]
Hi @rtau, given that homebrew ships pre-built binaries, we try to avoid adding options to our formulas unless they are very popular use cases. Thanks for the suggestion, but we are going to decline it at the moment, and continue to ship a full-featured xmrig.

**Comment by rtau** [comment]
Understood on shipping full feature formula, may I submit another pull request for copy the sample config.json?

**Comment by fxcoudert** [comment]
I've reopened, you can update to push the change limited to sample config.json
But I'd like to check: the project doesn't have a `make install` target?

**Comment by rtau** [comment]
Change on only adding `config.json` pushed.

No, the project do not have any `make install` target. 
Verified by manually running cmake on the source code.

**Comment by fxcoudert** [comment]
Thanks @rtau for the pull request!

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Pull request closed without merging**